src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -324,6 +355,38 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  override def getBucketUsage(projectName: RawlsBillingProjectName, bucketName: String, maxResults: Option[Long]): Future[BigInt] = {
+
+    def usageFromLogObject(o: StorageObject): Future[BigInt] = {
+      streamObject(o.getBucket, o.getName) { inputStream =>
+        val content = Source.fromInputStream(inputStream).mkString
+        val byteHours = BigInt(content.split('\n')(1).split(',')(1).replace("\"", ""))
+        // convert byte/hours to byte/days to better match the billing unit of GB/days
+        byteHours / 24
+      }
+    }
+
+    def recurse(pageToken: Option[String] = None): Future[BigInt] = {
```

Comment:
Given a max of a year's worth of storage logs, this should not be necessary (since the default page size is 1000). However, I developed this paging code in the previous implementation of the feature and I don't see any reason to throw it away. Also, if we can do this, then can simply decide some day to keep 5 years worth of logs and not worry about breaking this feature.